src/controllers/UserController.ts

```diff
@@ -14,31 +16,57 @@ export default class UserController {
    * @returns {Promise<Response>} Promise that resolves to an Express response
    */
   static async signup(req: Request, res: Response): Promise<Response> {
-    const { error } = userSchema.validate(req.body)
+    const { error } = userSchema.validate(req.body);
     if (error) {
-      return res.status(400).json({ message: error.details[0].message })
+      return res.status(400).json({ message: error.details[0].message });
     }
-    const salt = await bcrypt.genSalt(10)
-    const hashedpassword = await bcrypt.hash(req.body.password, salt)
 
-    const { email } = req.body
-    const existingUser = await User.findOne({ where: { email } })
+    const salt = await bcrypt.genSalt(10);
+    const hashedpassword = await bcrypt.hash(req.body.password, salt);
+
+    const { email } = req.body;
+    const existingUser = await User.findOne({ where: { email } });
     if (existingUser) {
-      return res.status(409).json({
-        message: 'This user already exists',
-      })
+      return res.status(409).json({ message: 'This user already exists' });
     }
+
     const newUser = new User({
       username: req.body.username,
       email: req.body.email,
       password: hashedpassword,
-    })
+    });
 
-    await newUser.save()
-    const token = jwt.sign({ id: newUser.id }, process.env.JWT_SECRET)
+    await newUser.save();
+    const token = jwt.sign({ id: newUser.id }, process.env.JWT_SECRET);
     return res.status(201).json({
       jwt: token,
       message: 'Account Created successfully',
-    })
+    });
+  }
+
+  /**
+   * Login method with Google OAuth
+   * @param {Request} req - Express request object
+   * @param {Response} res - Express response object
+   * @param {NextFunction} next - Express next middleware function
+   * @returns {void}
+   */
+  static loginWithGoogle(req: Request, res: Response, next: NextFunction): void {
+    passport.authenticate('google', (err: unknown, user: UserAttributes | null) => {
+      if (err) {
+        console.error('Internal Server Error:', err);
+        return res.status(500).json({ error: 'Internal Server Error' });
+      }
+      
+      if (!user) {
+        return res.status(401).json({ error: 'Authentication failed' });
+      }
+
+      const token = jwt.sign({ id: user.id }, process.env.JWT_SECRET as string, {
+        expiresIn: process.env.JWT_EXPIRATION as string,
+      });
```

Comment:
```suggestion
      const token = jwt.sign({ id: user.id }, process.env.JWT_SECRET as string, {
        expiresIn: process.env.JWT_EXPIRATION as string,
      });
```
this is can be moved somewhere like utils or helper that it can be reused